src/components/Miscellaneous/Badge/Badge.vue

```diff
@@ -1,18 +1,38 @@
 <template>
-  <div class="badge-container" :style="style">
-    <small :class="size">
+  <div
+    class="badge-container"
+    :style="style"
+  >
+    <PbIcon
+      v-if="hasIcon"
+      :icon="icon"
+      :class="iconClass"
+      class="pb-icon"
+    />
+    <small
+      :class="size"
+    >
       {{ title }}
     </small>
   </div>
 </template>
 
 <script>
 import { validateColor, isHexColor } from '@pb/utils/validator';
+import PbIcon from '@pb/Miscellaneous/Icon/Icon';
 
 export default {
   name: 'PbBadge',
 
+  components: {
+    PbIcon,
+  },
   props: {
+    icon: { type: String, default: '' },
+    hasIcon: {
```

Comment:
I think there is no necessity to pass a prop hasIcon, just pass the icon if you want e don't pass if you don't want

Reply:
I think that if you just don't give a value to :icon:"" it will display this without hasIcon prop
![noIcon](https://user-images.githubusercontent.com/51389447/168650213-6d0af4e1-8258-499b-8c49-5c765bdc9555.gif)

Reply:
Yes, but you can do the same way by just using `v-if="icon"` rather than `v-if="hasIcon"`

In this way, you can remove the hasIcon prop

Reply:
got it